### DISCUSSION TITLE
Wait X days before automerge PR

### DISCUSSION THREAD

**Discussion by sumanapotturu** [renovatebot/renovate#22166] [Suggest an Idea]
### Which Renovate are you using? CLI, App, or Pro
App

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps
GitHub

### What is your question?
We have a requirement to wait for X days before automerging any PR created through renovate bot. Is there a way to do this through config settings.

**Comment by rarkins** [comment]
Yes try adding [stabilityDays](https://docs.renovatebot.com/configuration-options/#stabilitydays) to your config and that should achieve what you are asking.

**Comment by sumanapotturu** [comment]
I tried that earlier but did not work. PR was merged immediately.
Docs mention it works if an update has a release date/timestamp. I would like to delay PR merge irrespective of the release date/timestamp

Below is the config I am using
```
{
  "packageRules": [
    {
      "packagePatterns": ["*"],
      "enabled": false
    },
    {
      "paths": ["renovateTestDir/package.json" ],
      "labels": [ "security" ],
      "packageNames": ["axios"],
      "allowedVersions": "0.18.1",
      "enabled": true
    },
     {
      "paths": ["renovateTestDir/package.json"],
      "labels": ["security"],
      "packageNames": ["sinon"],
      "allowedVersions": "7.5.0",
      "enabled": true,
      "stabilityDays": 2
    }
  ],
  "automerge": true,
  "automergeType": "pr",
   "stabilityDays": 3,
  "automergeComment": "Automerge",
  "branchPrefix": "security/renovate/",
  "prCreation": "immediate",
  "lockFileMaintenance": true,
  "rebaseStalePrs": true
}
```

Also tried this
```
{
  "packageRules": [
    {
      "packagePatterns": ["*"],
      "enabled": false
    },
    {
      "paths": ["renovateTestDir/package.json" ],
      "labels": [ "security" ],
      "packageNames": ["axios"],
      "allowedVersions": "0.18.1",
      "enabled": true
    },
     {
      "paths": ["renovateTestDir/package.json"],
      "labels": ["security"],
      "packageNames": ["sinon"],
      "allowedVersions": "7.5.0",
      "enabled": true
    }
  ],
  "automerge": true,
  "automergeType": "pr",
   "stabilityDays": 3,
  "automergeComment": "Automerge",
  "branchPrefix": "security/renovate/",
  "prCreation": "immediate",
  "lockFileMaintenance": true,
  "rebaseStalePrs": true
}
```

**Comment by rarkins** [comment]
I think this would require a feature addition to the main repo. To clarify, is this how you want it?

1. Renovate creates PRs immediately
2. PRs optionally will be configured with a "wait X days" option
3. Automerge will only apply after the X days is over

And the above will be independent of when the dependency itself was released, i.e. you just want the PR to have the delay regardless of whether the new version is recent or not?

**Comment by sumanapotturu** [comment]
Yes, that's right. This would give us a heads up and let us take appropriate actions before the PR is automerged.

**Comment by rarkins** [comment]
How would you propose to "stop" the automerge if you decide you don't want it automerged?

**Comment by sumanapotturu** [comment]
That's a great question. I believe today there is no config setting which would let us do this. 
Adding the X days would leave enough time to at least close PR manually. This will not work very well if "recreateClosed" is set to true.

One option I can think of is adding configurable "Block-Automerge" label. Merge only if X days have passed since PR create date and "Block-Automerge" label is set to false.
